lib/rust/ensogl/core/src/display/shape/primitive/glsl/shape.glsl

```diff
@@ -338,6 +362,21 @@ Shape inverse (Shape s1) {
 }
 
 Shape unify (Shape s1, Shape s2) {
+    if (input_display_mode == DISPLAY_MODE_CACHED_SHAPES_TEXTURE) {
+        // In DISPLAY_MODE_CACHED_SHAPES_TEXTURE As the color is not faded outside the shapes, we
+        // need to fade the foregroud color at some points before blending, otherwise it would
+        // "cover" the background color.
+        // There are two conditions we want to met:
+        // * We always want to fade it if outside foreground shape, but inside the background shape
+        //   (and the fg shape boundary over bg shape shall be anti-aliased, basing on sdf
+        //   information).
+        // * We want to keep the color consistent near border of the both shapes.
+        // We can fade the foreground shape always when the sdf distance is greated than the
+        // background shape - this way we met both conditions above.
```

Comment:
I would not understand this comment if I was not involved in creation of that. I don't know how to improve It though. Maybe better explanation of what "faded outside" is and why it is happening would be helpful? Maybe some ASCII art? I don't really know, but I feel that currently this comment is not as explicit as it should be. This part of the code is super hard and super hacky and it should be well documented.